gstreamer-sdp/src/m_i_k_e_y_message.rs

```diff
@@ -0,0 +1,206 @@
+use std::mem;
+use std::ptr;
+
+use ffi;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+
+use auto::{
+    MIKEYCacheType,
+    MIKEYMapType,
+    MIKEYPRFFunc,
+    MIKEYPayload,
+    MIKEYPayloadType,
+};
+use m_i_k_e_y_decrypt_info::MIKEYDecryptInfo;
+use m_i_k_e_y_encrypt_info::MIKEYEncryptInfo;
+use m_i_k_e_y_map_s_r_t_p::MIKEYMapSRTP;
+
+glib_wrapper! {
+    pub struct MIKEYMessage(Boxed<ffi::GstMIKEYMessage>);
+
+    match fn {
+        copy => |ptr| gobject_ffi::g_boxed_copy(ffi::gst_mikey_message_get_type(), ptr as *mut _) as *mut ffi::GstMIKEYMessage,
+        free => |ptr| gobject_ffi::g_boxed_free(ffi::gst_mikey_message_get_type(), ptr as *mut _),
+        get_type => || ffi::gst_mikey_message_get_type(),
+    }
+}
+
+impl MIKEYMessage {
+    pub fn new() -> MIKEYMessage {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_mikey_message_new())
+        }
+    }
+
+    pub fn new_from_bytes(bytes: &mut glib::Bytes, info: &mut MIKEYDecryptInfo, error: Option<Error>) -> MIKEYMessage {
+        unsafe {
+            let mut err = match error {
+                Some(err) => glib_ffi::g_error_new_literal(
```

Comment:
Wasn't sure how to convert a rust `Error` to a `glib::GError` so I ended up doing this... 😓

Reply:
This should actually be `pub fn new_from_bytes(bytes: &mut glib::Bytes, info: &mut MIKEYDecryptInfo) -> Result<MIKEYMessage, Error>`, like in other functions with `glib::Error`.

See e.g. here https://github.com/sdroege/gstreamer-rs/blob/b6a80b59fa0c32f210267de9fcd1d3adf69f1d9e/gstreamer/src/auto/functions.rs#L155-L157